File: tests/fixtures/ctxinfer_java_parent.scala
class Child(n: Int) extends ctxinfer_JavaParent("class=", n, n + 1)
object Values extends ctxinfer_JavaParent("object=", 4, 5)
object Empty extends ctxinfer_JavaParent("empty=")
object Spread extends ctxinfer_JavaParent("spread=", Array(6, 7): _*)
object Main {
  def main(args: Array[String]): Unit = {
    println(new Child(1).answer())
    println(Values.answer())
    println(Empty.answer())
    println(Spread.answer())
  }
}

File: tests/fixtures/ctxinfer_JavaParent.java
public class ctxinfer_JavaParent {
    private final String prefix;
    private final int[] values;
    public ctxinfer_JavaParent(String prefix, int... values) {
        this.prefix = prefix;
        this.values = values;
    }
    public String answer() {
        int sum = 0;
        for (int value : values) sum += value;
        return prefix + sum;
    }
}
